### PR TITLE
Update readme to latest template syntax

### PR DESCRIPTION
**Pull request by tpmccallum**
Signed-off-by: tpmccallum tim.mccallum@fermyon.com

This readme is also visible from https://crates.io/crates/spin-sdk

### CODE REVIEW

File: sdk/rust/README.md [hunk 1]
```diff
@@ -2,80 +2,105 @@
...
-- the function signature — `fn hello_world(req: Request) -> Result<Response>` —
-`req` can be any number of types including the built in `Request` type or 
-the `http::Request` from the popular `http` crate. Likewise, the response type
-can be many things including the built in `Response` type or the `http::Response` type
-from the `http` crate.
+- the `spin_sdk::http_component` macro marks the function as the entry point for the Spin component,
+- in the function signature (`fn handle_hello_world(req: Request) -> anyhow::Result<impl IntoResponse>`), `req` can be any number of types, including the built-in `Request` type or the `http::Request` from the popular `http` crate
+- in the function signature, the response type can be many things, including `anyhow::Result<impl IntoResponse>` (as shown above) or the `http::Response` type from the `http` crate (e.g. `Result<Response>`). Note - Using the `http` crate will require you to add it, e.g. `cargo install http`.
 
 ### Making outbound HTTP requests
```
Comment (tpmccallum) [inline sdk/rust/README.md:33]:
```suggestion
### Making Outbound HTTP Requests
```

File: sdk/rust/README.md [hunk 2]
```diff
@@ -2,80 +2,105 @@
...
+source = "target/wasm32-wasi/release/hello_world.wasm"
+allowed_outbound_hosts = ["https://random-data-api.fermyon.app"]
+[component.hello-world.build]
+command = "cargo build --target wasm32-wasi --release"
+watch = ["src/**/*.rs", "Cargo.toml"]
 ```
 
-Making a request to this component, we can see the appended header, and that the
-response contains the expected body:
+Spin build can be used to build all components defined in the Spin manifest file at the same time, and also has a flag that starts the application after finishing the compilation, `spin build --up`:
```
Comment (tpmccallum) [inline sdk/rust/README.md:83]:
```suggestion
## Building and Running the Spin Application

Spin build can be used to build all components defined in the Spin manifest file at the same time, and also has a flag that starts the application after finishing the compilation, `spin build --up`:
```

File: sdk/rust/README.md [hunk 3]
```diff
@@ -2,80 +2,107 @@
...
+source = "target/wasm32-wasi/release/hello_world.wasm"
+allowed_outbound_hosts = ["https://random-data-api.fermyon.app"]
+[component.hello-world.build]
+command = "cargo build --target wasm32-wasi --release"
+watch = ["src/**/*.rs", "Cargo.toml"]
 ```
 
-Making a request to this component, we can see the appended header, and that the
-response contains the expected body:
+## Building and Running the Spin Application
```
Comment (tpmccallum) [inline sdk/rust/README.md:83]:
```suggestion
### Building and Running the Spin Application
```

File: sdk/rust/README.md [hunk 4]
```diff
@@ -2,80 +2,107 @@
...
 
-- the `spin_sdk::http_component` macro — this marks the function as the
-  entrypoint for the Spin component
-- the function signature — `fn hello_world(req: Request) -> Result<Response>` —
-`req` can be any number of types including the built in `Request` type or 
-the `http::Request` from the popular `http` crate. Likewise, the response type
-can be many things including the built in `Response` type or the `http::Response` type
-from the `http` crate.
+- the `spin_sdk::http_component` macro marks the function as the entry point for the Spin component,
+- in the function signature (`fn handle_hello_world(req: Request) -> anyhow::Result<impl IntoResponse>`), `req` can be any number of types, including the built-in `Request` type or the `http::Request` from the popular `http` crate
```
Comment (rylev) [inline sdk/rust/README.md:30]:
```suggestion
- in the function signature (`fn handle_hello_world(req: Request) -> anyhow::Result<impl IntoResponse>`), `req` can be any number of types, including the built-in `Request` type or the `http::Request` type from the popular `http` crate
```

File: sdk/rust/README.md [hunk 5]
```diff
@@ -2,80 +2,107 @@
...
+- in the function signature, the response type can be many things, including `anyhow::Result<impl IntoResponse>` (as shown above) or the `http::Response` type from the `http` crate (e.g. `Result<Response>`). Note - Using the `http` crate will require you to add it, e.g. `cargo install http`.
```
Comment (rylev) [inline sdk/rust/README.md:31]:
```suggestion
- in the function signature, the response type can be anything that implements `IntoResponse` meaning the return type can any number of things including `anyhow::Result<impl IntoResponse>` (as shown above), `impl IntoResponse`, `Response`, `anyhow::Result<Response>`, or even the `http::Response` type from the `http` crate. 
  - Note: Using the `http` crate will require you to add it to your Cargo.toml manifest (i.e., `cargo add http`).
```

File: sdk/rust/README.md [hunk 6]
```diff
@@ -2,80 +2,107 @@
...
 
-### Making outbound HTTP requests
+### Making Outbound HTTP Requests
 
-Let's see an example where the component makes an outbound HTTP request to a
-server, modifies the result, then returns it:
+Let's see an example where the component makes an outbound HTTP request to a server, modifies the result, and then returns it:
 
 ```rust
+use anyhow::Result;
```
Comment (rylev) [inline sdk/rust/README.md:38]: We don't import `anyhow::Result` in the example above so I would remove this and change line 45 to:

```rust
async fn handle_hello_world(_req: Request) -> anyhow::Result<impl IntoResponse> {
```

File: sdk/rust/README.md [hunk 7]
```diff
@@ -2,80 +2,107 @@
...
-`req` can be any number of types including the built in `Request` type or 
-the `http::Request` from the popular `http` crate. Likewise, the response type
-can be many things including the built in `Response` type or the `http::Response` type
-from the `http` crate.
+- the `spin_sdk::http_component` macro marks the function as the entry point for the Spin component,
+- in the function signature (`fn handle_hello_world(req: Request) -> anyhow::Result<impl IntoResponse>`), `req` can be any number of types, including the built-in `Request` type or the `http::Request` from the popular `http` crate
+- in the function signature, the response type can be many things, including `anyhow::Result<impl IntoResponse>` (as shown above) or the `http::Response` type from the `http` crate (e.g. `Result<Response>`). Note - Using the `http` crate will require you to add it, e.g. `cargo install http`.
 
-### Making outbound HTTP requests
+### Making Outbound HTTP Requests
```
Comment (rylev) [inline sdk/rust/README.md:33]: It feels strange that we go into so much detail about outbound HTTP request but we don't mention any other SDK APIs like llm, key_value, sqlite, etc.

**Review by rylev — CHANGES_REQUESTED** [review]

File: sdk/rust/README.md [hunk 8]
```diff
@@ -2,80 +2,108 @@
...
 
-Let's see an example where the component makes an outbound HTTP request to a
-server, modifies the result, then returns it:
+Let's see an example where the component makes an outbound HTTP request to a server, modifies the result, and then returns it:
 
 ```rust
+use anyhow::Result;
```
Comment (tpmccallum) [inline sdk/rust/README.md:39]:
```suggestion
```

File: sdk/rust/README.md [hunk 9]
```diff
@@ -2,80 +2,107 @@
...
+$ curl -i localhost:3000
 HTTP/1.1 200 OK
-content-length: 29350
-content-type: text/html; charset=utf-8
-server: spin/0.1.0
+content-length: 77
+content-type: application/json
+
+{"timestamp":1702599575198,"fact":"Sharks lay the biggest eggs in the world"}
 ```
```
Comment (tpmccallum) [inline sdk/rust/README.md:108]:
```suggestion
```

## Fermyon Developer Home

For many more examples, please see the official [Fermyon Developer Home](https://developer.fermyon.com/). 

[The language support overview page](https://developer.fermyon.com/spin/v2/language-support-overview) lists all of Spin's features (key-value storage, SQLite, MySQL, Redis, Serverless AI etc.) and links to how each feature is implemented in a given language (Rust, TS/JS, Python, TinyGo).
```

**Review by tpmccallum — COMMENTED** [review]
Hi @rylev 
Agreed, the README file is limited in what it explains to the user. 
I have added a section that links to the developer documentation.

File: sdk/rust/README.md (hunk 6 above)
Comment (tpmccallum) [reply sdk/rust/README.md]: Done!

File: sdk/rust/README.md (hunk 7 above)
Comment (tpmccallum) [reply sdk/rust/README.md:34]: Agreed, I have added a link to the developer documentation so readers have somewhere to go, when they finish reading this README file.

**Comment by rylev** [comment]
@tpmccallum I don't see the section linking to the developer docs. Where can I find that?

File: sdk/rust/README.md [hunk 10]
```diff
@@ -2,80 +2,106 @@
 
 The Spin Rust SDK makes it easy to build Spin components in Rust.
```
Comment (tpmccallum) [inline sdk/rust/README.md:4]:
```suggestion

## Fermyon Developer Home

This `README` file provides a few examples, such as writing Spin HTTP components in Rust and making outbound HTTP requests. For comprehensive information, visit the official [Fermyon Developer Home](https://developer.fermyon.com/). This resource includes [a page on installing Spin](https://developer.fermyon.com/spin/v2/install#installing-spin), [a quickstart guide](https://developer.fermyon.com/spin/v2/quickstart), and [a language support overview page](https://developer.fermyon.com/spin/v2/language-support-overview). The latter lists all of Spin's features—including key-value storage, SQLite, MySQL, Redis, Serverless AI, etc.—and their implementation in specific languages such as Rust, TS/JS, Python, and TinyGo.

```

**Comment by tpmccallum** [comment]
Oh, my apologies @rylev (and good catch). It seems the backticks stopped the text from being an inline commit (and my changes were just added as general comments).
Screenshot 2023-12-20 at 08 48 50
This is probably a blessing in disguise because it made me rethink where that paragraph should go for impact. I have placed the links to the developer documentation at the top of the file. Please approve if you like this approach so we can merge. Also, I am not sure how to make sure that these new changes are rendered on crates.io page (from memory, sometimes a new version bump and commit/merge to the crate is required, i.e. `0.0.N`)

File: sdk/rust/README.md [hunk 11]
```diff
@@ -2,80 +2,110 @@
...
+
+Once our application is running, we can make a request (by visiting `http://localhost:3000/` in a web browser) or using `curl` as shown below:
+
+```bash
+$ curl -i localhost:3000
 HTTP/1.1 200 OK
-content-length: 29350
-content-type: text/html; charset=utf-8
-server: spin/0.1.0
-```
```
Comment (rylev) [inline sdk/rust/README.md:81]: I think these backticks need to be brought back.

**Review by rylev — APPROVED** [review]
Left one comment with something we should address but other than that looks great!

File: sdk/rust/README.md (hunk 11 above)
Comment (tpmccallum) [reply sdk/rust/README.md:81]: Agreed, thanks, great catch.

**Comment by tpmccallum** [comment]
Thanks so much for the review @rylev ; greatly appreciated!!!

**Pull request merged**